### PR TITLE
Dataset test tweaks

### LINKED ISSUE #13014: AWS Aurora Postgresql: test failures with `lein test`
**Describe the bug**
I'm running the test suite of master against AWS Aurora Postgresql (11.6). I just provisioned a new database instance and accepted the defaults. I've set the MB_DB_* values to point at the primary (`postgres`) user that AWS created. I'm running the tests to see if we can use Aurora Postgres as the **application** database (not the warehouse).

Most of the tests are passing, except what's pasted below in logs. It's unclear to me whether this is an issue with the test suite or if there's something about Aurora Postgresql that is incompatible for use as the application database.

This is my first time running these tests

**Logs**
```
ERROR in (failure-test) (dataset_test.clj:34)
POST /api/dataset
Even if a query fails we still expect a 202 response from the API
Saved QueryExecution
expected: (= {:hash true, :database_id (mt/id), :result_rows 0, :started_at true, :executor_id (mt/user->id :rasta), :native true, :pulse_id nil, :id true, :card_id nil, :context :ad-hoc, :error true, :row_count 0, :running_time true, :dashboard_id nil} (check-error-message (format-response (most-recent-query-execution))))
INFO data.env :: Running QP tests against these drivers: #{:h2}
  actual: clojure.lang.ExceptionInfo: Expected results to be a map! Got: null
{:results nil}
 at metabase.api.dataset_test$format_response.invokeStatic (dataset_test.clj:34)
    metabase.api.dataset_test$format_response.invoke (dataset_test.clj:32)
    metabase.api.dataset_test$fn__84958.invokeStatic (dataset_test.clj:146)
    metabase.api.dataset_test/fn (dataset_test.clj:96)
    clojure.test$test_var$fn__9737.invoke (test.clj:717)
    clojure.test$test_var.invokeStatic (test.clj:717)
    clojure.test$test_var.invoke (test.clj:708)
    clojure.test$test_vars$fn__9763$fn__9768.invoke (test.clj:735)
    clojure.test$default_fixture.invokeStatic (test.clj:687)
    clojure.test$default_fixture.invoke (test.clj:683)
    clojure.test$test_vars$fn__9763.invoke (test.clj:735)
    clojure.test$default_fixture.invokeStatic (test.clj:687)
    clojure.test$default_fixture.invoke (test.clj:683)
    clojure.test$test_vars.invokeStatic (test.clj:731)
    clojure.test$test_all_vars.invokeStatic (test.clj:737)
    clojure.test$test_ns.invokeStatic (test.clj:758)
    clojure.test$test_ns.invoke (test.clj:743)
    user$eval11211$fn__11272.invoke (form-init2279900357599966159.clj:1)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.AFn.applyTo (AFn.java:144)
    clojure.core$apply.invokeStatic (core.clj:667)
    clojure.core$apply.invoke (core.clj:660)
    leiningen.core.injected$compose_hooks$fn__11141.doInvoke (form-init2279900357599966159.clj:1)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invokeStatic (core.clj:665)
    clojure.core$apply.invoke (core.clj:660)
    leiningen.core.injected$run_hooks.invokeStatic (form-init2279900357599966159.clj:1)
    leiningen.core.injected$run_hooks.invoke (form-init2279900357599966159.clj:1)
    leiningen.core.injected$prepare_for_hooks$fn__11146$fn__11147.doInvoke (form-init2279900357599966159.clj:1)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.lang.AFunction$1.doInvoke (AFunction.java:31)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    clojure.core$map$fn__5866.invoke (core.clj:2755)
    clojure.lang.LazySeq.sval (LazySeq.java:42)
    clojure.lang.LazySeq.seq (LazySeq.java:51)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.next (RT.java:713)
    clojure.core$next__5386.invokeStatic (core.clj:64)
    clojure.core$reduce1.invokeStatic (core.clj:944)
    clojure.core$reduce1.invokeStatic (core.clj:934)
    clojure.core$merge_with.invokeStatic (core.clj:3059)
    clojure.core$merge_with.doInvoke (core.clj:3051)
    clojure.lang.RestFn.applyTo (RestFn.java:139)
    clojure.core$apply.invokeStatic (core.clj:667)
    clojure.test$run_tests.invokeStatic (test.clj:768)
    clojure.test$run_tests.doInvoke (test.clj:768)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invokeStatic (core.clj:665)
    clojure.core$apply.invoke (core.clj:660)
    user$eval11211$fn__11284$fn__11317.invoke (form-init2279900357599966159.clj:1)
    user$eval11211$fn__11284$fn__11285.invoke (form-init2279900357599966159.clj:1)
    user$eval11211$fn__11284.invoke (form-init2279900357599966159.clj:1)
    user$eval11211.invokeStatic (form-init2279900357599966159.clj:1)
    user$eval11211.invoke (form-init2279900357599966159.clj:1)
    clojure.lang.Compiler.eval (Compiler.java:7177)
    clojure.lang.Compiler.eval (Compiler.java:7167)
    clojure.lang.Compiler.load (Compiler.java:7636)
    clojure.lang.Compiler.loadFile (Compiler.java:7574)
    clojure.main$load_script.invokeStatic (main.clj:475)
    clojure.main$init_opt.invokeStatic (main.clj:477)
    clojure.main$init_opt.invoke (main.clj:477)
    clojure.main$initialize.invokeStatic (main.clj:508)
    clojure.main$null_opt.invokeStatic (main.clj:542)
    clojure.main$null_opt.invoke (main.clj:539)
    clojure.main$main.invokeStatic (main.clj:664)
    clojure.main$main.doInvoke (main.clj:616)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.lang.Var.applyTo (Var.java:705)
    clojure.main.main (main.java:40)
load-data for h2 test-data-with-null-date-checkins users took 31.1 ms
load-data for h2 test-data-with-null-date-checkins categories took 6.2 ms
load-data for h2 test-data-with-null-date-checkins venues took 23.2 ms
load-data for h2 test-data-with-null-date-checkins checkins took 139.5 ms
Sync :h2 Database test-data-with-null-date-checkins took 4.0 s

lein test :only metabase.api.dataset-test/basic-query-test

FAIL in (basic-query-test) (dataset_test.clj:80)
POST /api/dataset
Just a basic sanity check to make sure Query Processor endpoint is still working correctly.
Saved QueryExecution
expected: {:hash true,
           :database_id 1,
           :result_rows 1,
           :started_at true,
           :executor_id 2,
           :native false,
           :pulse_id nil,
           :id true,
           :card_id nil,
           :context :ad-hoc,
           :error nil,
           :row_count 1,
           :running_time true,
           :dashboard_id nil}
  actual: {:hash true,
           :database_id 57,
           :result_rows 0,
           :started_at true,
           :executor_id 2,
           :native false,
           :pulse_id nil,
           :id true,
           :card_id nil,
           :context :ad-hoc,
           :error "You do not have permissions to run this query.",
           :row_count 0,
           :running_time true,
           :dashboard_id nil}
    diff: - {:row_count 1, :error nil, :result_rows 1, :database_id 1}
          + {:row_count 0, :error "You do not have permissions to run this query.", :result_rows 0, :database_id 57}

...

Ran 2456 tests containing 7833 assertions.
1 failures, 1 errors.
Tests failed.
Error encountered performing task 'test' with profile(s): 'base,system,user,provided,dev,with-include-drivers-middleware,test-common'
Tests failed.
```

**To Reproduce**
Set environment variables pointing to AWS Aurora Postgresql 11.6 database instance.

**Expected behavior**
`lein test` passes without errors

**Screenshots**
None

**Information about your Metabase Installation:**
Not an install, just trying to run the tests. Head of master as of filing this ticket: 5f358727f59a73c253978c76f7983222c0ae443f

**Severity**
Low severity.

**Additional context**
I'm evaluating Aurora Postgres to see if it can be used as the **application** database.

### PR DESCRIPTION
**Pull request by walterl**
Fixes #13014.

Please see commit messages for more details.

### CODE REVIEW

**Review by ascandella — COMMENTED** [review]
I'm not quite qualified to review this at the moment since I've never touched this codebase, I'll leave it to one of the experts

File: test/metabase/api/dataset_test.clj [hunk 1]
```diff
@@ -48,7 +52,9 @@
        :else
        [k v]))))
 
-(defn- most-recent-query-execution [] (db/select-one QueryExecution {:order-by [[:id :desc]]}))
+(defn- most-recent-query-execution []
+  (Thread/sleep 100)
+  (db/select-one QueryExecution {:order-by [[:id :desc]]}))
```
Comment (sbelak) [inline test/metabase/api/dataset_test.clj:57]: Have you considered ordering by `started_at`? I'm really uneasy about using sleep here as it's a great recipe for intermittent failures.

Comment (walterl) [reply test/metabase/api/dataset_test.clj:57]: I'll try that :+1:.

Comment (walterl) [reply test/metabase/api/dataset_test.clj]: Looks like that did the trick :tada:

**Review by camsaul — APPROVED** [review]
LGTM 👍

**Pull request merged**